### DISCUSSION TITLE
Android: What's the inst parameter in setMorphWeights(inst, weights)?

### DISCUSSION THREAD

**Discussion by will4621d** [google/filament#4986] [Q&A] [answered]
I called modelViewer.asset.getFirstEntityByName("Head_Netural.001"), which returned an int value, then I passed it to SetMorphWeight, but nothing happened.
In the end, I tried setMorphWeights(1, weights) and  it worked. 
Btw, asset.getName(1) also return null.
Why this happened, can someone explain this to me? Thanks.

Model: https://github.com/will4621d/Sample/blob/main/head.zip

**Comment by bejado** [comment, marked as answer]
`getFirstEntityByName` returns an `Entity`. `setMorphWeights` takes a `RenderableManager::Instance` as its first argument. You can get an `Instance` from an `Entity` via the [`RenderableManager::getInstance(utils::Entity)` method](https://github.com/google/filament/blob/251abf0545b9b4afcc49a21de5620e79a743ddad/filament/include/filament/RenderableManager.h#L100).
